flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java

```diff
@@ -424,6 +425,33 @@ protected void initChannel(SocketChannel ch) {
 		LOG.info("Web frontend listening at " + address + ':' + port);
 	}
 
+	public static Archiver[] getArchivers() {
```

Comment:
i wanted to do something like this originally. However, since the archiving is done in the ``´MemoryArchivist``` i would have to pass the ```WebRuntimeMonitor``` reference to it. This is problematic since you now add an order in which they have to created. This specifically caused issues with the Yarn/Mesos runner that create the JobManager before the WebRuntimeMonitor.

It's a stupid problem and i agree that the static method approach is rather brittle, but i couldn't find a good solution for the above issue.